src/renderers/shared/fiber/ReactFiberCompleteWork.js

```diff
@@ -198,7 +206,33 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
       case FunctionalComponent:
         return null;
       case ClassComponent: {
-        // We are leaving this subtree, so pop context if any.
+        if (
+          workInProgress.effectTag & PerformedWork &&
+          !(workInProgress.effectTag & PreparedWork)
```

Comment:
In what scenarios can both `workInProgress.effectTag & PerformedWork` and `workInProgress.effectTag & PreparedWork` be true?

Reply:
Names are a bit confusing. `PerformedWork` means this fiber was worked on. `PreparedWork` means this fiber has stuff that needs to be garbage collected if it's dropped without committing.

Reply:
They're separate because not all fibers that are worked on need to be garbage collected.